pose_estimation_pcl:
  ros__parameters:
    general:
      processing_period_ms: 100
      goicp_debug : false
      save_debug_clouds : true
      voxel_size: 0.01
      object_frame: "docking_st"

    registration:
      use_goicp : false
      gicp_fitness_threshold: 0.05

    clustering:
      cluster_tolerance: 0.05 #in meters, default is 0.03
      min_cluster_size: 150 #250
      max_cluster_size: 75000 #50000

    3d_decriptors:
      visualize_normals: false
      normal_radius: 0.1 # the highger the more neighbor to consider, improving the normal vector
      fpfh_radius: 0.07 # the highger the more neighbor to consider, improving the descriptor
      similarity_threshold : 0.6 #0.5
    
    plane_detection:
      plane_distance_threshold: 0.01
      max_plane_iterations: 100
      min_plane_points: 800
      max_planes: 3